CorsixTH/CMakeLists.txt

```diff
@@ -55,6 +55,13 @@ else()
   set(CORSIX_TH_OS "unix")
 endif()
 
+# Report system architecture in compile_opts.arch
+if(MSVC)
+  set(CORSIX_TH_ARCH ${CMAKE_CXX_COMPILER_ARCHITECTURE_ID})
```

Comment:
On discord I checked / pointed out that CMAKE_CXX_COMPILER_ARCHITECTURE_ID is CMake 3.10+ only. I don't mind bumping the cmake_minimum_version, or version guarding this condition.